services/common/src/components/reports/AddReportDetails.tsx

```diff
@@ -0,0 +1,476 @@
+import { Alert, Button, Col, Form, Input, List, Popconfirm, Row, Typography } from "antd";
+import React, { FC, useEffect, useState } from "react";
+import ArrowRightOutlined from "@ant-design/icons/ArrowRightOutlined";
+import { Link } from "react-router-dom";
+import {
+  getDropdownMineReportCategoryOptions,
+  getMineReportDefinitionOptions,
+} from "@mds/common/redux/selectors/staticContentSelectors";
+import PropTypes from "prop-types";
+import { ReportSubmissions } from "@mds/common/components/reports/ReportSubmissions";
+import { compose } from "redux";
+import { useDispatch, connect, useSelector } from "react-redux";
+import { Field, reduxForm, formValueSelector } from "redux-form";
+import * as FORM from "@mds/common/constants/forms";
+import { dateNotInFuture, required, yearNotInFuture } from "@mds/common/redux/utils/Validate";
+import { renderConfig } from "@mds/common/components/common/config";
+import { flatMap, uniqBy } from "lodash";
+import Callout from "@mds/common/components/common/Callout";
+import ReportsTable from "./ReportsTable";
+import {
+  resetForm,
+  formatComplianceCodeValueOrLabel,
+  createDropDownList,
+  sortListObjectsByPropertyLocaleCompare,
+} from "@mds/common/redux/utils/helpers";
+import CustomPropTypes from "@mds/common/customPropTypes";
+import moment from "moment";
+import { closeModal, openModal } from "@mds/common/redux/actions/modalActions";
+
+import { getMineReports } from "@mds/common/redux/selectors/reportSelectors";
+import {
+  createMineReport,
+  fetchMineReports,
+  updateMineReport,
+} from "@mds/common/redux/actionCreators/reportActionCreator";
+import { modalConfig } from "@mds/common/components/modalContent/config";
+import TextArea from "antd/lib/input/TextArea";
+
+const selector = formValueSelector(FORM.ADD_REPORT);
+
+const updateMineReportDefinitionOptions = (
+  mineReportDefinitionOptions,
+  selectedMineReportCategory
+) => {
+  let mineReportDefinitionOptionsFiltered = mineReportDefinitionOptions;
+
+  if (selectedMineReportCategory) {
+    mineReportDefinitionOptionsFiltered = mineReportDefinitionOptions.filter(
+      (rd) =>
+        rd.categories.filter((c) => c.mine_report_category === selectedMineReportCategory).length >
+        0
+    );
+  }
+
+  let dropdownMineReportDefinitionOptionsFiltered = createDropDownList(
+    mineReportDefinitionOptionsFiltered,
+    "report_name",
+    "mine_report_definition_guid"
+  );
+  dropdownMineReportDefinitionOptionsFiltered = sortListObjectsByPropertyLocaleCompare(
+    dropdownMineReportDefinitionOptionsFiltered,
+    "label"
+  );
+};
+
+interface AddReportDetailsProps {
+  mineGuid: string;
+  updateMineReportSubmissions: any;
```

Comment:
For anything coming from the `mapDispatchToProps` we can add a type like this:

`updateMineReportSubmissions: ActionCreator<typeof updateMineReportSubmissions>;`

This will give you all the correct typing when working with it throughout the file and make it easier to maintain.

Reply:
Updated.